File: JavaScript/7js 2/number.js
/*
|--------------------------------------------------------------------------
| Number of People in the Bus
|--------------------------------------------------------------------------
|
*/

function number(busStops) {
   let x = 0;
   let y = 0
   for (let i = 0; i < busStops.length; i++) {
      x += busStops[i][0];
      y += busStops[i][1];
   }
   return x - y;
}


const number = busStops => busStops.reduce((people, stop) => people + stop[0] - stop[1], 0)


const number = function (busStops) {
   return busStops.map(x => x[0] - x[1]).reduce((x, y) => x + y);
}


const number = (busStops) => busStops.reduce((rem, [on, off]) => rem + on - off, 0);


const number = function (busStops) {
   let peopleInBus = 0;
   busStops.forEach(([enter, left]) => {
      peopleInBus += enter - left;
   })
   return peopleInBus
}


console.log(number([
   [10, 0],
   [3, 5],
   [5, 8]
]), 5);
console.log(number([
   [3, 0],
   [9, 1],
   [4, 10],
   [12, 2],
   [6, 1],
   [7, 10]
]), 17);
console.log(number([
   [3, 0],
   [9, 1],
   [4, 8],
   [12, 2],
   [6, 1],
   [7, 8]
]), 21);

File: Python/7py 2/number.py
"""
|--------------------------------------------------------------------------
| Number of People in the Bus
|--------------------------------------------------------------------------
|
"""

def number(bus_stops):
   x = 0
   y = 0
   for i in range(len(bus_stops)):
      x += bus_stops[i][0]
      y += bus_stops[i][1]
   return x - y


def number(bus_stops):
   return sum(on - off for on, off in bus_stops)


def number(bus_stops):
   sum=0
   for i in bus_stops:
      sum+=i[0]-i[1]
   return sum


def number(bus_stops):
   return sum([item[0] - item[1] for item in bus_stops]) 


def number(bus_stops):
   return sum(map(lambda x: x[0]-x[1],bus_stops))


print(number([[10,0],[3,5],[5,8]]),5)
print(number([[3,0],[9,1],[4,10],[12,2],[6,1],[7,10]]),17)
print(number([[3,0],[9,1],[4,8],[12,2],[6,1],[7,8]]),21)